app/models/aggregated_cash_income.rb

```diff
@@ -25,4 +25,12 @@ def update(params)
 
     save_cash_transaction_records
   end
+
+  def cash_income_options
+    if Setting.enhanced_bank_upload?
+      legal_aid_application.regular_transactions.credits.map(&:transaction_type)
+    else
+      legal_aid_application.transaction_types.not_children.credits
```

Comment:
Why does this line not work with regular payments already? I thought that transaction types were created and added to the application alongside regular payments when these are added on the previous page. I am still trying to understand that part of the code though so this might be an incorrect understanding?

Reply:
A `LegalAidApplication` only `has_many :transaction_types, through: :legal_aid_application_transaction_types` (i.e the join table).

`RegularTransaction` effectively models the same relationship between a `LegalAidApplication` and a `TransactionType`, except the relationship contains more information and thereby models the domain more literally.

Does that make sense?